plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1751,6 +1755,124 @@ public void testOptimizeUsingForcedPartitioning()
         assertThat(getAllDataFilesFromTableDirectory(tableName)).isEqualTo(union(initialFiles, updatedFiles));
     }
 
+    @Test
+    public void testRegisterTableWithTableLocation()
+    {
+        String tableName = "test_register_table_with_table_location_" + randomTableSuffix();
+
+        assertUpdate(format("CREATE TABLE %s (a int, b varchar, c boolean)", tableName));
+        assertUpdate(format("INSERT INTO %s values(1, 'INDIA', true)", tableName), 1);
+        assertUpdate(format("INSERT INTO %s values(2, 'USA', false)", tableName), 1);
+
+        String tableLocation = getTableLocation(tableName);
+        String showCreateTableOld = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
+        // Drop table from metastore and use the same table to register again
+        dropTableFromMetastore(tableName);
+
+        assertUpdate("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+        String showCreateTableNew = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
+
+        assertThat(showCreateTableOld).isEqualTo(showCreateTableNew);
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES " +
+                        "ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true'), " +
+                        "ROW(INT '2', VARCHAR 'USA', BOOLEAN 'false')");
+        assertUpdate(format("DROP TABLE %s", tableName));
```

Comment:
nit: We prefer `+` to `format` method in this case. It provides SQL syntax highlight in IDE. 
https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#prefer-string-formatting
> f you only need to append something, consider using the + operator.